core-common/src/main/java/org/mapstruct/Mapper.java

```diff
@@ -112,4 +112,15 @@
      * @return The strategy to be applied when {@code null} is passed as source value to the methods of this mapper.
      */
     NullValueMappingStrategy nullValueMappingStrategy() default NullValueMappingStrategy.DEFAULT;
+
+    /**
+     * The strategy to use for applying {@code @Mapping} configurations of prototype methods in the interface specified
```

Comment:
That comes from writing the javadoc before actually diving into the implementation - because as it is now, it should somehow work for iterables and maps as well. 
Tests for them are also on my list.